kroxylicious-test-tools/src/main/java/io/kroxylicious/test/tester/KroxyliciousTesterBuilder.java

```diff
@@ -0,0 +1,96 @@
+    private DefaultKroxyliciousTester.ClientFactory clientFactory = (clusterName, defaultConfiguration) -> new KroxyliciousClients(defaultConfiguration);
+    private ConfigurationBuilder configurationBuilder;
+
+    private Function<String, ConfigurationBuilder> mockConfigurationFunction;
+
+    private List<ResponsePayload> mockPayloads = new ArrayList<>();
+
+    public KroxyliciousTesterBuilder setKroxyliciousFactory(Function<Configuration, AutoCloseable> kroxyliciousFactory) {
+        this.kroxyliciousFactory = kroxyliciousFactory;
+        return this;
+    }
+
+    KroxyliciousTesterBuilder setClientFactory(DefaultKroxyliciousTester.ClientFactory clientFactory) {
+        this.clientFactory = clientFactory;
+        return this;
+    }
+
+    public KroxyliciousTesterBuilder setConfigurationBuilder(ConfigurationBuilder configurationBuilder) {
+        this.configurationBuilder = configurationBuilder;
+        return this;
+    }
+
+    public KroxyliciousTesterBuilder setTrustStoreLocation(String trustStoreLocation) {
+        this.trustStoreLocation = trustStoreLocation;
+        return this;
+    }
+
+    public KroxyliciousTesterBuilder setTrustStorePassword(String trustStorePassword) {
+        this.trustStorePassword = trustStorePassword;
+        return this;
+    }
+
+    public KroxyliciousTesterBuilder setMockConfigurationFunction(Function<String, ConfigurationBuilder> mockConfigurationFunction) {
+        this.mockConfigurationFunction = mockConfigurationFunction;
+        return this;
+    }
+
+    public KroxyliciousTesterBuilder addMockResponse(ResponsePayload responsePayload) {
+        this.mockPayloads.add(responsePayload);
+        return this;
+    }
+
+    public KroxyliciousTester createDefaultKroxyliciousTester() {
+        final TrustStoreConfiguration trustStoreConfiguration = trustStoreLocation != null ? new TrustStoreConfiguration(trustStoreLocation, trustStorePassword) : null;
+        return new DefaultKroxyliciousTester(configurationBuilder, kroxyliciousFactory, clientFactory, trustStoreConfiguration);
+    }
+
+    /**
+     * Creates a kroxylicious tester for a kroxylicious instance that is proxying a
+     * mock kafka broker. The mock can be configured to respond with an exact API
+     * message, and be used to verify what was sent to it.
+     * @return MockServerKroxyliciousTester
+     */
+    public MockServerKroxyliciousTester createMockKroxyliciousTester() {
+        final MockServer mockServer = MockServer.startOnRandomPort();
+        for (ResponsePayload mockPayload : mockPayloads) {
+            mockServer.addMockResponseForApiKey(mockPayload);
+        }
+        return new MockServerKroxyliciousTester(mockServer, mockConfigurationFunction);
+    }
+
+    public KroxyliciousTester createTester() {
+        if (mockPayloads.isEmpty()) {
+            return createDefaultKroxyliciousTester();
+        }
+        else {
+            return createMockKroxyliciousTester();
+        }
+    }
```

Comment:
This currently doesn't work as most usages still want to call `addMockResponseForApiKey` or `dropWhen`. So maybe this method should be 
```suggestion
    <T extends KroxyliciousTester> public T createTester() {
        if (mockPayloads.isEmpty()) {
            return createDefaultKroxyliciousTester();
        }
        else {
            return createMockKroxyliciousTester();
        }
    }
```